File: spec/lua/test_to_string_custom.lua
local luaunit = require("luaunit")

require("to_string_custom")

TestToStringCustom = {}

function TestToStringCustom:test_term_strz()
    local r = ToStringCustom:from_file("src/term_strz.bin")

    luaunit.assertEquals(tostring(r), "s1 = foo, s2 = bar")
end

File: spec/java/src/io/kaitai/struct/spec/TestToStringCustom.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.ToStringCustom;
import org.testng.annotations.Test;
import static org.testng.Assert.*;
public class TestToStringCustom extends CommonSpec {
    @Test
    public void testToStringCustom() throws Exception {
        ToStringCustom r = ToStringCustom.fromFile(SRC_DIR + "term_strz.bin");

        assertEquals(r.toString(), "s1 = foo, s2 = bar");
    }
}
